### PR TITLE
feat(twap): display banner with swap amount difference

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Spec: https://docs.google.com/presentation/d/1ujFpnYbZ2eeaPA3v8DXRctWnDHrdlkx-63qScYrwlVk/edit#slide=id.g25b719a5370_0_65

| -- | -- |
|--------|--------|
| image | image | 

@fairlighteth please, fix styles of the banner

### CODE REVIEW

**Comment by shoom3301** [comment]
@anxolin could you give me a text for the info icon on the banner please?

File: src/modules/twap/containers/TwapFormWarnings/warnings/SwapPriceDifferenceWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        <div>
+          {isTwapBetter ? (
+            <>
+              You are expected to receive <TokenAmount amount={amount} tokenSymbol={amount.currency} /> more compared to
+              the same SWAP order
+            </>
+          ) : (
+            <>
+              Given the current Market Fees (<FiatAmount amount={feeFiatAmount} /> per chunk), it’s better to reduce the
+              number of parts or use SWAP
```
Comment (alfetopito) [inline src/modules/twap/containers/TwapFormWarnings/warnings/SwapPriceDifferenceWarning.tsx:44]: Link to SWAP?

**Review by alfetopito — APPROVED** [review]

File: src/modules/twap/containers/TwapFormWarnings/warnings/SwapPriceDifferenceWarning.tsx (hunk 1 above)
Comment (shoom3301) [reply src/modules/twap/containers/TwapFormWarnings/warnings/SwapPriceDifferenceWarning.tsx]: Thanks! added

**Comment by elena-zh** [comment]
Hey @shoom3301 , great job!
My notes are: 
1. Styles (addressed in #2939 )
2. No tooltip content in the warning
3. Maybe we should set a min threshold for displaying this banner ([e.g.1% of a trade](https://cowservices.slack.com/archives/C0361CDG8GP/p1690371873881739?thread_ts=1690290151.789909&cid=C0361CDG8GP))?
4. I think it would be great to add a [condition ](https://cowservices.slack.com/archives/C0361CDG8GP/p1690372785669789?thread_ts=1690290151.789909&cid=C0361CDG8GP)to the 2nd warning like: 
 - 2 parts: remove '`to reduce the number of parts`' from the warning
 - >2 parts: leave it as it is. 

LMK please if I need to create a separate task for any of these points.

**Comment by shoom3301** [comment]
>Maybe we should set a min threshold for displaying this banner ([e.g.1% of a trade](https://cowservices.slack.com/archives/C0361CDG8GP/p1690371873881739?thread_ts=1690290151.789909&cid=C0361CDG8GP))?

@elena-zh added the condition. Set the threshold 0.5%

**Review by elena-zh — APPROVED** [review]
Changes LGTM!

**Review by anxolin — COMMENTED** [review]
Great PR!

File: src/modules/tradeQuote/hooks/useSetTradeQuoteParams.ts [hunk 2]
```diff
@@ -6,14 +6,14 @@ import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
...
 
-  const amount = currencyAmount?.quotient.toString() || null
-
+  const amountStr = amount?.quotient.toString()
   useEffect(() => {
     updateTradeQuote({ response: null, error: null })
     updateState({ amount })
-  }, [amount, updateTradeQuote, updateState])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [amountStr, updateTradeQuote, updateState])
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply src/modules/tradeQuote/hooks/useSetTradeQuoteParams.ts:18]: Yes, really, I forgot about it. Thanks!

[the remark this answers is not in the document]
Comment (shoom3301) [reply src/modules/tradeQuote/hooks/useSetTradeQuoteParams.ts:18]: Fixed: https://github.com/cowprotocol/cowswap/commit/774af471c0f8cafcc5ca14d7312b9569f7e06da1

[the remark this answers is not in the document]
Comment (anxolin) [reply src/modules/tradeQuote/hooks/useSetTradeQuoteParams.ts:18]: Thanks!!

**Pull request merged**